### DISCUSSION TITLE
Finding the right device

### DISCUSSION THREAD

**Discussion by Radio-biscuit** [savonet/liquidsoap#2981] [Q&A]
Hello,

Getting to grips with Liquidsoap, and while it's a challenge I'm getting some good results.

The question today is that I use Voicemeeter on my WIndows PC to control my audio sources and outputs. I have used the --list-portaudio-devices option to get a list of devices, but of course this keeps changing every time I restart my computer. It's not hard to get a working device, but I want to find out how to interpret the information from the listing option:

What is the host API? This is just a number (indicating the subsystem), but I have no way of finding out what number is for what system? Which one is kernel streaming? Which one is ASIO? Which one is WDM/WASAPI etc? I'd really love to know a bit more about this so that I can automate the selection of the device ID.

Alternatively is there a way of connecting to a named output? The names don't change, so how can I do this instead?

Any advice will be gratefully received while I continue to manually set up after each restart ...

Thanks,

Andy

**Comment by toots** [comment]
Hi! 

`--list-portaudio-devices` is tapping directly into the portaudio internals for which I/we have very little knowledge. I imagine that this question should be better directed at the windows port of portaudio, unfortunately.

**Reply by Radio-biscuit** [reply]
Ah, okay. I have no idea about portaudio either! Thanks for the heads up.

I'm now learning about Jack on Linux, so maybe I don't have to worry so much about portaudio (but it's still there on Linux too).
